project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,137 @@
  */
 package akka
 
-import sbt._
-import sbt.Keys._
 import com.typesafe.tools.mima.plugin.MimaKeys.reportBinaryIssues
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies.ModuleId
+import sbt.Keys._
+import sbt._
+
+import scala.collection.immutable
 
 object ValidatePullRequest extends AutoPlugin {
 
+  val ValidatePR = config("ValidatePR") extend Test
+
+  /*
+    Assumptions:
+      Env variables set "by Jenkins" are assumed to come from this plugin:
+      https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin
+   */
+
+  // settings
+  val TargetBranchEnvVarName = "PR_TARGET_BRANCH"
+  val TargetBranchJenkinsEnvVarName = "ghprbTargetBranch"
+  val sourceBranch = settingKey[String]("Branch containing the changes of this PR")
+
+  val SourceBranchEnvVarName = "PR_SOURCE_BRANCH"
+  val SourceBranchJenkinsEnvVarName = "ghprbSourceBranch"
+  val targetBranch = settingKey[String]("Branch with which the PR changes should be diffed against")
+
+  // determining touched dirs and projects
+  val changedDirectories = taskKey[immutable.Set[String]]("List of touched modules in this PR branch")
+  val projectIsAffectedByChanges = taskKey[Boolean]("True if this project is affected by the PR and should be rebuilt")
+
+  // running validation
   val validatePullRequest = taskKey[Unit]("Additional tasks for pull request validation")
 
   override def trigger = allRequirements
 
   override lazy val projectSettings = Seq(
-    validatePullRequest := (),
-    validatePullRequest <<= validatePullRequest.dependsOn(test in Test),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.PerformanceTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.LongRunningTest"),
```

Comment:
I think we have marked a lot with `LongRunningTest`, e.g. cluster tests. Perhaps we should remove that marker for some important tests?